Update deploy-jobs.md for self-deferral (#4023)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/deploy-jobs.md

    - **Environment Variables** &mdash; Define [environment variables](/docs/build/environment-variables) to customize the behavior of your project when the deploy job runs.
    - **Target Name** &mdash; Define the [target name](/docs/build/custom-target-names) to customize the behavior of your project when the deploy job runs. Environment variables and target names are often used interchangeably. 
    - **Run Timeout** &mdash; Cancel the deploy job if the run time exceeds the timeout value. 

>     - **Compare changes against** — By default, it’s set to **No deferral**. Select either **Environment** or **This Job**  to let dbt Cloud know what it should compare the changes against.  

Reviewer: ```suggestion
    - **Compare changes against** &mdash; By default, it’s set to **No deferral**. Select either **Environment** or **This Job**  to let dbt Cloud know what it should compare the changes against.  
```

>     - **Compare changes against ** option — By default, it’s set to **No deferral**. For Deploy jobs, you can select either no deferral, deferral to an environment, or self defer (to the same job).

Reviewer: ```suggestion
    - **Compare changes against** — By default, it’s set to **No deferral**. Select either **Environment** or **This Job**  to let dbt Cloud know what it should compare the changes against.  
```

>     - **Compare changes against** &mdash; By default, it’s set to **No deferral**. Select either **Environment** or **This Job**  to let dbt Cloud know what it should compare the changes against.  

Reviewer: ```suggestion
    - **Compare changes against** &mdash; By default, it’s set to **No deferral**. Select either **Environment** or **This Job** to let dbt Cloud know what it should compare the changes against.  
```
